Реализовал localstorage для корзины товаров

Rubsee opened the pull request:
Реализовал:
- localstorage для корзины товаров
- Закрытие корзины по нажатию на область экрана вне корзины

Rubsee committed:
Реализовал localstorage для корзины товаров

Реализовал:
- localstorage для корзины товаров
- Закрытие корзины по нажатию на область экрана вне корзины

Rubsee committed:
Исправил недочет

Исправил последовательность key, value в localstorage

captain-zsa commented on source/js/productCart.js:
```diff
@@ -24,20 +27,44 @@ const removeProductFromCart = (productId) => {
-};
\ No newline at end of file
+    addToStorage('cart', product);
+    cartCount.textContent = cartList.childElementCount;
```
я бы советовал чутка иначе. Чтобы вам каждый раз не искать ваш этот cartList (а это согласитесь будет совсем неудобно делать если вдруг такого меню просто не будет и количество товаров нужно будет отображать совсем в другом месте) – лучше просто посчитать, сколько товаров прямо сейчас внутри localStorage. Подумайте как это реализовать и попробуйте)) Это гораздо удобнее будет, чем ориентироваться на количество элементов в разметке.

captain-zsa commented on source/js/productCart.js:
```diff
@@ -24,20 +27,44 @@ const removeProductFromCart = (productId) => {
+    getStorage('cart').forEach(product => {
+        addProductToCart(product);
+    });
+    cartCount.textContent = cartList.childElementCount;
```
Собственно, аналогично.

captain-zsa reviewed (commented):
Решается буквально 1-й строчкой всё. Если успешно получится – можно ДЗ сливать))

Rubsee committed:
Работа над ошибками

Исправил cartCount

Rubsee committed:
Работа над ошибками

- исправил счетчик товаров в корзине при удалении. (раньше при удалении счетчик отставал на еденицу)
- исправил увеличение счетчика при обновлении страницы
